templates/nextjs/src/components/Layout.tsx

```diff
@@ -117,12 +204,36 @@ export const Layout = ({ children }: { children: React.ReactNode }) => {
 
           <Link href="/faucet">Faucet</Link>
 
+          {isConnected && (
+            <Button onClick={disconnect}>Disconnect Wallet</Button>
+          )}
+          {!isConnected && <Button onClick={connect}>Connect Wallet</Button>}
+
+          {shouldShowAddNetworkButton && (
+            <Button onClick={tryToAddNetwork} className="bg-red-500">
+              Wrong Network
+            </Button>
+          )}
+
           <div className="ml-auto">
-            <BurnerWallet />
+            {connectedWallet ? (
+              <WalletDisplay
+                wallet={connectedWallet}
+                walletBalance={connectedWalletBalance}
+              />
+            ) : (
+              <BurnerWallet />
```

Comment:
Given that you've changed `BurnerWallet` to be `WalletDisplay` as well, you could use the `WalletDisplay` component directly here as well and get rid of the `BurnerWallet` component altogether.
```jsx
<div className="ml-auto">
  {connectedWallet ? (
    <WalletDisplay
      wallet={connectedWallet}
      walletBalance={connectedWalletBalance}
    />
  ) : (
    <WalletDisplay
      wallet={burnerWallet}
      walletBalance={connectedWalletBalance}
    />
  )}
</div>
```

You can even use the `AppContext` inside of `WalletDisplay` to get `connectedWallet` and `burnerWallet` and have this code just be:
```jsx
<div className="ml-auto">
    <WalletDisplay />
</div>
```